includes/sanitizers/class-amp-story-export-sanitizer.php

```diff
@@ -0,0 +1,168 @@
+<?php
+/**
+ * Class AMP_Story_Export_Sanitizer.
+ *
+ * @package AMP
+ */
+
+/**
+ * Class AMP_Story_Export_Sanitizer
+ *
+ * Sanitizes AMP Stories during export.
+ *
+ * @since 1.2.1
+ */
+class AMP_Story_Export_Sanitizer extends AMP_Base_Sanitizer {
+
+	/**
+	 * Default args.
+	 *
+	 * @var array
+	 */
+	protected $DEFAULT_ARGS = [
+		'base_url' => '',
+	];
+
+	/**
+	 * Default assets.
+	 *
+	 * @var array
+	 */
+	protected $assets = [];
+
+	/**
+	 * Sanitize the AMP Story.
+	 */
+	public function sanitize() {
+
+		// AMP Story Node.
+		$this->sanitize_assets(
+			'amp-story',
+			[
+				'publisher-logo-src',
+				'publisher',
+				'poster-portrait-src',
+				'poster-square-src',
+				'poster-landscape-src',
+			]
+		);
+
+		// AMP Image Nodes.
+		$this->sanitize_assets(
+			'amp-img',
+			[
+				'src',
+				'srcset',
+			]
+		);
+
+		// AMP Video Nodes.
+		$this->sanitize_assets(
+			'amp-video',
+			[
+				'src',
+				'poster',
+			]
+		);
+
+		// Send the raw HTTP header for the export assets.
+		if ( ! empty( $this->assets ) && ! headers_sent() ) {
+			header( 'X-AMP-Export-Assets: ' . implode( ',', $this->assets ) );
+		}
+	}
+
+	/**
+	 * Add and sanitize the export assets for the AMP Story.
+	 *
+	 * @param string $name       The DOMElement name.
+	 * @param array  $attributes The DOMElement attributes.
+	 */
+	public function sanitize_assets( $name, $attributes ) {
+		$nodes     = $this->dom->getElementsByTagName( $name );
+		$num_nodes = $nodes->length;
+
+		// Verify we have a value to update the paths with.
+		$update_path = ! empty( $this->args['base_url'] ) ? $this->args['base_url'] : false;
+
+		/**
+		 * Generates the new asset path.
+		 *
+		 * @param string $asset The original URL.
+		 *
+		 * @return bool|string Returns false when $update_path is false, else the new URL.
+		 */
+		$get_asset_path = function( $asset ) use ( $update_path ) {
+			if ( $asset && $update_path ) {
+				$args = [
+					$update_path,
+					'assets',
+					basename( $asset ),
+				];
+				return implode( '/', $args );
+			}
+			return false;
+		};
+
+		if ( 0 < $num_nodes ) {
+			for ( $i = $num_nodes - 1; $i >= 0; $i-- ) {
+				$node = $nodes->item( $i );
+
+				if ( ! $node instanceof DOMElement ) {
+					continue;
+				}
+
+				if ( ! empty( $attributes ) ) {
```

Comment:
To reduce nesting, this could be:

```
if ( empty( $attributes ) ) {
    return;
}
```

But actually, shouldn't that be moved up to the top of the method? But is it even actually needed since it is always going to be an array? And looping over an empty array just won't do anything.